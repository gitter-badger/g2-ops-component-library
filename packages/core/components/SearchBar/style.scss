.searchTypesMenuButton {
  margin: 0 -10px 0 -10px;
  font-size: 13px;
  width: max-content;
}

.prefixContextualMenu {
  padding: 0px;
}

.searchBarDiv {
  display: flex;
}

.searchBarTextField {
  min-width: 25em;
}

.searchBarCheckbox {
  margin-left: 1em;
}
