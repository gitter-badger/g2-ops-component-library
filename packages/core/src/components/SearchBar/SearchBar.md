SearchBar Example:

```js
var searchTypes = [
	{ key: 'lot', name: 'Lot' },
  { key: 'seller', name: 'Seller' },
  { key: 'owner', name: 'Owner' },
  { key: 'seller-personnel', name: 'Seller Personnel' },
  { key: 'facility', name: 'Facility' },
  { key: 'location', name: 'Location' },
  { key: 'buyer', name: 'Buyer' },
];

<div style={{ width: 'max-content', border: '1px solid #a6a6a6' }}>
  <SearchBar
    searchType={{ key: 'lot', name: 'Lot' }}
    searchTypes={searchTypes}
    borderless
    searchText='Default Search Text'
    handleSearch={(result) => console.log(result,'Search Clicked')}
    showCheckbox={true}
    themeVariant='dark'
  />
</div>
```

```js
var searchTypes = [
	{ key: 'lot', name: 'Lot' },
  { key: 'seller', name: 'Seller' },
  { key: 'owner', name: 'Owner' },
  { key: 'seller-personnel', name: 'Seller Personnel' },
  { key: 'facility', name: 'Facility' },
  { key: 'location', name: 'Location' },
  { key: 'buyer', name: 'Buyer' },
];

<div style={{ padding: '5px',width: '100%', border: '1px solid #a6a6a6',backgroundColor: 'indigo' }}>
  <SearchBar
    searchType={{ key: 'lot', name: 'Lot' }}
    searchTypes={searchTypes}
    borderless
    searchText='Default Search Text'
    handleSearch={(result) => console.log(result,'Search Clicked')}
    showCheckbox={true}
    themeVariant='light'
  />
</div>
```
