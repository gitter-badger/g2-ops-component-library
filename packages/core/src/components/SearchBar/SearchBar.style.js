const lightColorStyle = { color: '#fff' }
const darkColorStyle = { color: '#000'}
const getTextColor = (themeVariant) => (themeVariant === 'dark'? darkColorStyle: lightColorStyle)
const darkStyles = {
  checkboxCheckedHovered: {
    borderColor: '#000',
    background: '#fff',
  },
  checkboxCheckedFocused: {
    background: '#fff',
    borderColor: '#000',
  },
  checkboxChecked: {
    background: '#fff',
    borderColor: '#000',
  },
  checkmark: {
    opacity: '0',
    color: 'black',
  },
  checkmarkChecked: {
    opacity: '1',
  },
  text: {
    color: '#000',
    fontSize: '12px',
  },
  textHovered: {
    color: '#000'
  },
  textFocused: {
    color: '#000'
  },
}
const lightStyles = {
  checkboxCheckedHovered: {
    borderColor: '#fff',
    background: '#fff',
  },
  checkboxCheckedFocused: {
    background: '#fff',
    borderColor: '#fff',
  },
  checkbox: {
    background: '#fff',
  },
  checkboxChecked: {
    background: '#fff',
    borderColor: '#fff',
  },
  checkmark: {
    opacity: '0',
    color: '#000',
  },
  checkmarkChecked: {
    opacity: '1',
    color: '#000',
  },
  text: {
    color: '#fff',
    fontSize: '12px',
  },
  textHovered: {
    color: '#fff'
  },
  textFocused: {
    color: '#fff'
  },
}
export const checkboxStyle = (themeVariant) => themeVariant==='dark'? darkStyles: lightStyles

export default {}
