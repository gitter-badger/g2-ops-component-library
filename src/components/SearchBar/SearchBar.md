SearchBar Example:

```js
<div style={{ width: 'max-content', border: '1px solid #a6a6a6' }}>
  <SearchBar
    searchTypeValue={{ key: 'lot', name: 'Lot' }}
    borderless
    showCheckbox={false}
  />
</div>
```