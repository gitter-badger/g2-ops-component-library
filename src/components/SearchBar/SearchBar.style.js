const colorStyle = { color: '#fff' }

export const checkboxStyle = {
  checkboxCheckedHovered: {
    borderColor: 'black',
  },
  checkboxCheckedFocused: {
    background: '#fff',
    borderColor: '#fff',
  },
  checkboxChecked: {
    background: '#fff',
    borderColor: '#fff',
  },
  checkmark: {
    opacity: '0',
    color: 'black',
  },
  checkmarkChecked: {
    opacity: '1',
  },
  text: {
    ...colorStyle,
    fontSize: '12px',
  },
  textHovered: colorStyle,
  textFocused: colorStyle,
}

export default {}
