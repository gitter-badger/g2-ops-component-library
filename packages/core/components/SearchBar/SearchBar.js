// @flow
import React, { PureComponent } from 'react'
import renderIf from 'render-if'
import { TextField } from 'office-ui-fabric-react/lib/TextField'
import { DirectionalHint } from 'office-ui-fabric-react/lib/ContextualMenu'
import { DefaultButton, IconButton } from 'office-ui-fabric-react/lib/Button'

import { Checkbox } from 'components/Checkbox/Checkbox'

import wrapFabricContext from '../../wrapFabricContext'
import { checkboxStyle } from './SearchBar.style'

import './style.scss'

type SearchType = {
  key: string,
  name: string,
}

type SearchBarPropType = {
  /** Search Type, one of the items of type { key,name } from the searchTypes provided  */
  searchType: SearchType,
  /** Search Types to be rendered in the searchMenu on the left */
  searchTypes: Array<SearchType>,
  /** Trigger search handler */
  handleSearch(): any,
  /** search text to display within the Search box */
  searchText: string,
  /** boolean to show Checkbox */
  showCheckbox: boolean,
  /** Callback triggered when the search type changes */
  onSearchTypeChange?: SearchType => any,
}

type SearchBarStateType = {
  searchType: SearchType,
}

class SearchBar extends PureComponent<SearchBarPropType, SearchBarStateType> {
  state = {
    searchType: this.props.searchType || { key: 'lot', name: 'Lot' },
  }

  handleMenuChange = (event: SyntheticMouseEvent<HTMLInputElement>, item: SearchType) => {
    this.setState({
      searchType: item,
    })
    if (this.props.onSearchTypeChange) {
      this.props.onSearchTypeChange(item)
    }
  }

  renderSearchBarMenu = () => {
    const { searchTypes } = this.props
    return (
      <DefaultButton
        className="searchTypesMenuButton"
        text={this.state.searchType.name}
        menuProps={{
          isBeakVisible: false,
          directionalHint: DirectionalHint.bottomLeftEdge,
          directionalHintForRTL: DirectionalHint.bottomLeftEdge,
          gapSpace: 0,
          beakWidth: 20,
          directionalHintFixed: true,
          onItemClick: this.handleMenuChange,
          items: searchTypes,
        }}
      />
    )
  }

  renderSearchIcon = () => (
    <IconButton onClick={this.props.handleSearch} iconProps={{ iconName: 'search' }} title={'Search'} />
  )

  render() {
    const { searchType: searchTypeProp, showCheckbox, ...searchBarProps } = this.props
    const { searchType } = this.state
    const renderAllFacilitiesCheckbox = renderIf(searchType.key === 'lot' && showCheckbox)
    return (
      <div className="searchBarDiv">
        <TextField
          {...searchBarProps}
          placeholder="Type here to Search"
          className="searchBarTextField"
          onRenderPrefix={this.renderSearchBarMenu}
          onRenderSuffix={this.renderSearchIcon}
        />
        {renderAllFacilitiesCheckbox(
          <Checkbox className="searchBarCheckbox" label={'All Facilities'} styles={checkboxStyle} />,
        )}
      </div>
    )
  }
}

export default SearchBar
