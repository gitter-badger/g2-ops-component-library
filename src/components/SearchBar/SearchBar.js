import React from 'react'
import PropTypes from 'prop-types'
import renderIf from 'render-if'
import { TextField } from 'office-ui-fabric-react/lib/TextField'
import { DirectionalHint } from 'office-ui-fabric-react/lib/ContextualMenu'
import { DefaultButton, IconButton } from 'office-ui-fabric-react/lib/Button'
import { SearchBox } from 'office-ui-fabric-react/lib/SearchBox'
import Checkbox from '../Checkbox/Checkbox'
import { wrapFabricContext } from '../../wrapFabricContext'
import './style.scss'

const defaultSearchTypes = [
  { code: 'lot',
    description: 'Lot',
  },
  { code: 'seller',
    description: 'Seller',
  },
  { code: 'owner',
    description: 'Owner',
  },
  { code: 'location',
    description: 'Location',
  },
  { code: 'facility',
    description: 'Facility',
  },
  { code: 'sellerPersonnel',
    description: 'Seller Personnel',
  },
]

class SearchBar extends React.Component {
  static propTypes = {
    handleMenuChange: PropTypes.func,
    searchTypevalue: PropTypes.string,
    searchTypes: PropTypes.arrayOf(PropTypes.shape),
    handleSearch: PropTypes.func,
    showCheckbox: PropTypes.bool
  }

  state = {
    searchTypeValue: this.props.searchTypeValue || { key: 'lot', name: 'Lot' },
  }

  handleMenuChange = (event, index, value) => {
    this.setState({
      searchTypeValue: value
    }, () => {
      if(this.props.handleMenuChange) {
        this.props.handleMenuChange(event, index, value)
      }
    })
  }

  renderContextualMenu = () => (
    <DefaultButton
      className="searchTypesMenuButton"
      text={this.state.searchTypeValue.name}
      menuProps={{
        isBeakVisible: false,
        directionalHint: DirectionalHint.bottomLeftEdge,
        directionalHintForRTL: DirectionalHint.bottomLeftEdge,
        gapSpace: 0,
        beakWidth: 20,
        directionalHintFixed: true,
        onItemClick: (event, item) => this.handleMenuChange(event,null,item),
        items: [
          { key: 'lot', name: 'Lot' },
          { key: 'seller', name: 'Seller' },
          { key: 'owner', name: 'Owner' },
          { key: 'seller-personnel', name: 'Seller Personnel' },
          { key: 'facility', name: 'Facility' },
          { key: 'location', name: 'Location' },
          { key: 'buyer', name: 'Buyer' },
        ]
      }}
    />
  )

  renderSearchIcon = (handleSearch) => (
    <IconButton
      onClick={handleSearch}
      iconProps={ { iconName: 'search' } }
      title='Search'
    />
  )

  render() {
    const { searchTypeValue: searchTypeValueProps, searchTypes = defaultSearchTypes, handleSearch, showCheckbox, ...searchBarProps } = this.props
    const { searchTypeValue } = this.state
    const renderAllFacilitiesCheckbox = renderIf(searchTypeValue.key === 'lot')
    const colorStyle = { color: '#fff' }
    return (
      <div className="searchBarDiv">
        <TextField
          {...searchBarProps}
          placeholder="Type here to Search"
          className="searchBarTextField"
          onRenderPrefix={this.renderContextualMenu}
          onRenderSuffix={() => this.renderSearchIcon(handleSearch)}
        />
        {showCheckbox && renderAllFacilitiesCheckbox(
          <Checkbox
            className="searchBarCheckbox"
            label={'All Facilities'}
            styles={{
              checkboxCheckedHovered: {
                borderColor: 'black',
              },
              checkboxCheckedFocused: {
                background: '#fff',
                borderColor: '#fff'
              },
              checkboxChecked: {
                background: '#fff',
                borderColor: '#fff'
              },
              checkmark: {
                opacity: '0',
                color: 'black'
              },
              checkmarkChecked: {
                opacity: '1'
              },
              text: colorStyle,
              textHovered: colorStyle,
              textFocused: colorStyle
            }}
          />
        )}
      </div>
    )
  }
}

export default SearchBar
