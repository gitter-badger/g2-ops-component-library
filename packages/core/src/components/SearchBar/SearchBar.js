// @flow
import React, { PureComponent } from 'react'
import renderIf from 'render-if'
import { TextField } from 'office-ui-fabric-react/lib/TextField'
import { DirectionalHint } from 'office-ui-fabric-react/lib/ContextualMenu'
import { DefaultButton, IconButton } from 'office-ui-fabric-react/lib/Button'

import { Checkbox } from 'components/Checkbox/Checkbox'

import { wrapFabricContext } from 'utilities/wrapFabricContext'
import { checkboxStyle } from './SearchBar.style'

import './style.scss'

type SearchType = {
  key: string,
  name: string,
}

type SearchBarPropType = {
  /** Search Type, one of the items of type { key,name } from the searchTypes provided  */
  searchType: SearchType,
  /** Search Types to be rendered in the searchMenu on the left */
  searchTypes: Array<SearchType>,
  /** Trigger search handler */
  handleSearch: ({
    searchType: SearchType,
    searchText: string,
    allFacilitiesChecked: boolean|null
  }) => mixed,
  /** search text to display within the Search box */
  searchText: string,
  /** boolean to show Checkbox */
  showCheckbox: boolean,
  /** Callback triggered when the search type changes */
  onSearchTypeChange?: SearchType => any,
  /** Callback invoked when the search text changes */
  onSearchTextChange?: string => any,
  /* light or dark */
  themeVariant: 'light'| 'dark',
}

type SearchBarStateType = {
  searchType: SearchType,
  searchText: string,
  allFacilitiesChecked: boolean | null,
}

class SearchBar extends PureComponent<SearchBarPropType, SearchBarStateType> {
  state = {
    searchType: this.props.searchType || { key: 'lot', name: 'Lot' },
    searchText: this.props.searchText || '',
    allFacilitiesChecked: null,
  }
  handleSearchTextChange = (text: string) => {
    this.setState({
      searchText: text
    }, () => {
      if(this.props.onSearchTextChange){
        this.props.onSearchTextChange(text)
      }
    })

  }
  handleMenuChange = (event: SyntheticMouseEvent<HTMLInputElement>, item: SearchType) => {
    this.setState({
      searchType: item,
    })
    if (this.props.onSearchTypeChange) {
      this.props.onSearchTypeChange(item)
    }
  }

  handleSearchIconClick = () => {
    const { searchType, searchText, allFacilitiesChecked} = this.state
    this.props.handleSearch({searchType, searchText, allFacilitiesChecked})
  }
  handleFacilityChanged = (e, value: boolean) => {
    this.setState({
      allFacilitiesChecked: value
    })
  }
  renderSearchBarMenu = () => {
    const { searchTypes } = this.props
    return (
      <DefaultButton
        className="searchTypesMenuButton"
        text={this.state.searchType.name}
        menuProps={{
          isBeakVisible: false,
          directionalHint: DirectionalHint.bottomLeftEdge,
          directionalHintForRTL: DirectionalHint.bottomLeftEdge,
          gapSpace: 0,
          beakWidth: 20,
          directionalHintFixed: true,
          onItemClick: this.handleMenuChange,
          items: searchTypes,
        }}
      />
    )
  }

  renderSearchIcon = () => (
    <IconButton onClick={this.handleSearchIconClick} iconProps={{ iconName: 'search' }} title={'Search'} />
  )

  render() {
    const { searchType: searchTypeProp, showCheckbox,themeVariant, ...searchBarProps } = this.props
    const { searchType, searchText,allFacilitiesChecked } = this.state
    const renderAllFacilitiesCheckbox = renderIf(searchType.key === 'lot' && showCheckbox)
    return (
      <div className="searchBarDiv">
        <TextField
          {...searchBarProps}
          placeholder="Type here to Search"
          className="searchBarTextField"
          onRenderPrefix={this.renderSearchBarMenu}
          onRenderSuffix={this.renderSearchIcon}
          value={searchText}
          onChanged={this.handleSearchTextChange}
        />
        {renderAllFacilitiesCheckbox(
          <Checkbox
            className="searchBarCheckbox"
            label="All Facilities"
            styles={checkboxStyle(themeVariant)}
            value={allFacilitiesChecked}
            onChange={this.handleFacilityChanged}
          />,
        )}
      </div>
    )
  }
}

export { SearchBar }
