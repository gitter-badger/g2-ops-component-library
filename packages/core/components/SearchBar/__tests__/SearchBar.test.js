import React from 'react'
import renderer from 'react-test-renderer'

import {SearchBar} from '../SearchBar'

const searchTypes = [
  { key: 'lot', name: 'Lot' },
  { key: 'seller', name: 'Seller' },
  { key: 'owner', name: 'Owner' },
  { key: 'seller-personnel', name: 'Seller Personnel' },
  { key: 'facility', name: 'Facility' },
  { key: 'location', name: 'Location' },
  { key: 'buyer', name: 'Buyer' },
]

describe('<SearchBar />', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <SearchBar
        searchType={{ key: 'lot', name: 'Lot' }}
        searchTypes={searchTypes}
        borderless
        searchText="Default Search Text"
        handleSearch={() => console.log('Search Clicked')}
        showCheckbox={false}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
  test('should render allFacilitiesCheckbox when searchType = "lot" & showCheckbox="true"', () => {
    const tree = renderer.create(
      <SearchBar
        searchType={{ key: 'lot', name: 'Lot' }}
        searchTypes={searchTypes}
        borderless
        searchText="Default Search Text"
        handleSearch={() => console.log('Search Clicked')}
        showCheckbox={false}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
