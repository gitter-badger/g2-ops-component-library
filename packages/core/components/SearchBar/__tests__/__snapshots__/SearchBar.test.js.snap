// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render allFacilitiesCheckbox when searchType = "lot" & showCheckbox="true" 1`] = `
<div
  className="searchBarDiv"
>
  <div
    className="ms-TextField root searchBarTextField ms-TextField--borderless rootIsBorderless"
  >
    <div
      className="ms-TextField-wrapper wrapper"
    >
      <div
        className="ms-TextField-fieldGroup fieldGroup"
      >
        <div
          className="ms-TextField-prefix fieldPrefixSuffix"
        >
          <button
            aria-describedby={null}
            aria-expanded={false}
            aria-haspopup={true}
            aria-label={undefined}
            aria-labelledby={null}
            aria-owns={null}
            aria-pressed={undefined}
            className="ms-Button searchTypesMenuButton ms-Button--default root-37"
            data-is-focusable={true}
            disabled={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <div
              className="ms-Button-flexContainer flexContainer-38"
            >
              <div
                className="ms-Button-textContainer textContainer-39"
              >
                <div
                  className="ms-Button-label label-41"
                  id="id__10"
                >
                  Lot
                </div>
              </div>
              <i
                aria-hidden={true}
                className="root-45 ms-Button-menuIcon menuIcon-42"
                data-icon-name="ChevronDown"
                role="presentation"
              />
            </div>
          </button>
        </div>
        <input
          aria-describedby={undefined}
          aria-invalid={false}
          aria-label={undefined}
          className="ms-TextField-field field"
          id="TextField8"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder="Type here to Search"
          type="text"
          value=""
        />
        <div
          className="ms-TextField-suffix fieldPrefixSuffix"
        >
          <button
            aria-describedby={null}
            aria-label={undefined}
            aria-labelledby={null}
            aria-pressed={undefined}
            className="ms-Button ms-Button--icon root-48"
            data-is-focusable={true}
            disabled={undefined}
            onClick={[Function]}
            title="Search"
            type="button"
          >
            <div
              className="ms-Button-flexContainer flexContainer-38"
            >
              <i
                aria-hidden={true}
                className="root-45 ms-Button-icon icon-40"
                data-icon-name="search"
                role="presentation"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SearchBar /> should render properly 1`] = `
<div
  className="searchBarDiv"
>
  <div
    className="ms-TextField root searchBarTextField ms-TextField--borderless rootIsBorderless"
  >
    <div
      className="ms-TextField-wrapper wrapper"
    >
      <div
        className="ms-TextField-fieldGroup fieldGroup"
      >
        <div
          className="ms-TextField-prefix fieldPrefixSuffix"
        >
          <button
            aria-describedby={null}
            aria-expanded={false}
            aria-haspopup={true}
            aria-label={undefined}
            aria-labelledby={null}
            aria-owns={null}
            aria-pressed={undefined}
            className="ms-Button searchTypesMenuButton ms-Button--default root-37"
            data-is-focusable={true}
            disabled={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <div
              className="ms-Button-flexContainer flexContainer-38"
            >
              <div
                className="ms-Button-textContainer textContainer-39"
              >
                <div
                  className="ms-Button-label label-41"
                  id="id__2"
                >
                  Lot
                </div>
              </div>
              <i
                aria-hidden={true}
                className="root-45 ms-Button-menuIcon menuIcon-42"
                data-icon-name="ChevronDown"
                role="presentation"
              />
            </div>
          </button>
        </div>
        <input
          aria-describedby={undefined}
          aria-invalid={false}
          aria-label={undefined}
          className="ms-TextField-field field"
          id="TextField0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder="Type here to Search"
          type="text"
          value=""
        />
        <div
          className="ms-TextField-suffix fieldPrefixSuffix"
        >
          <button
            aria-describedby={null}
            aria-label={undefined}
            aria-labelledby={null}
            aria-pressed={undefined}
            className="ms-Button ms-Button--icon root-48"
            data-is-focusable={true}
            disabled={undefined}
            onClick={[Function]}
            title="Search"
            type="button"
          >
            <div
              className="ms-Button-flexContainer flexContainer-38"
            >
              <i
                aria-hidden={true}
                className="root-45 ms-Button-icon icon-40"
                data-icon-name="search"
                role="presentation"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
